### ISSUE TITLE
Swagger documentation of related entities isn't proper

### ISSUE THREAD

**Issue by kate-kate** [api-platform/core#986]
Hello everyone. 
I have a Job entity with related Campaign. According to [https://api-platform.com/docs/core/serialization-groups-and-relations](url) I configured them like this:

```php
<?php

namespace AppBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use ApiPlatform\Core\Annotation\ApiResource;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * Job
 * @ApiResource(attributes={"normalization_context"={"groups"={"dashboard"}}})
 * @ORM\HasLifecycleCallbacks
 * @ORM\Table(name="job")
 * @ORM\Entity
 */
class Job
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @Groups({"dashboard"})
     * @ORM\Column(name="status", type="integer", nullable=false)
     */
    private $status;

    /**
     * @Groups({"dashboard"})
     * @ORM\Column(name="postcode", type="string", length=255, nullable=true)
     */
    private $postcode;

    /**
     * @Groups({"dashboard"})
     * @ORM\ManyToOne(targetEntity="Campaign")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="campaign_id", referencedColumnName="id")
     * })
     */
    private $campaign;
 
    // ...
}
```

```php
<?php
namespace AppBundle\Entity;

use ApiPlatform\Core\Annotation\ApiResource;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * Campaign
 * @ORM\HasLifecycleCallbacks
 * @ApiResource(attributes={"normalization_context"={"groups"={"dashboard"}}})
 * @ORM\Table(name="campaign")
 * @ORM\Entity
 */
class Campaign
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $id;

    /**
     * @var string
     * @ORM\Column(name="external_id", type="string", length=255, nullable=true)
     */
    private $externalId;

    /**
     * @var string
     * @Groups({"dashboard"})
     * @ORM\Column(name="name", type="string", length=255, nullable=false)
     */
    private $name; 

    // ...
}

```

And it works perfectly OK, when I get the response body, but the example value of response is quite strange.

Example value swagger shows to me
```json
[
  {
    "status": 0,
    "postcode": "string",
    "campaign": {
      "status": 0,
      "postcode": "string",
      "campaign": {}
    }
  }
]
```

As you can see, there are job fields expected from campaign instead of its own

**Comment by jordscream** [comment]
Hi @kate-kate ,

Generally, the same for me, I get several errors concerning "Example value" generation.
 
I recognize it's a hard (according to me) stuff to work on Swagger generation definition with API platform data ^^
I will start, at first, by strong functionnal tests, with a maximum of cases (groups, relations, embedded...)

@Simperfit , what I see [here](https://github.com/api-platform/core/blob/2.0/src/Swagger/Serializer/DocumentationNormalizer.php#L484) might be the problem. It takes the definitionkey Jobs instead of calling $this->resourceMetadataFactory->create($className)->getShortName().

But I suppose, if we change that, it will breaks another things ^^

**Comment by jonadoe** [comment]
Hi @kate-kate and @jordscream 

I also struggled with this and created PR https://github.com/api-platform/core/pull/996 which fixes this issue. But to be absolutely sure it would be greate if @Simperfit could also check the PR to make sure it doesn't break PR #888

**Comment by kate-kate** [comment]
@jonadoe I've tested your changes locally, and they solved not only this issue, but also #988 
Great job! Looking forward your changes will be merged to master
